src/client/src/components/DraggableSidebarItem/index.tsx

```diff
@@ -73,6 +80,17 @@ const DraggableSidebarItem = ({
   const handleCloseOnClick = () => {
     idx && handleCloseClick && handleCloseClick(idx - 1); // correction for idx + 1 to prevent 0th falsey behaviour
   };
+
+  const handleKeyDownInput = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const inputKeyCheck = /^[A-Za-z0-9_\- ]$/
+    let event = e.target as HTMLInputElement;
+    if (event.value.length === maxInputLength && inputKeyCheck.test(e.key)) {
```

Comment:
you see, you would think this would work, and  so did I.... but turns out that `onChange` only works when the input box actually changes. So if I am at the max input box and i hit a key, it doesn't register a change and wont run the code. Also I the CTRL key issue is by design, because its not an input character, you are not trying to input a new character when pressing ctrl so the error shouldn't come up. 

I tried to use the onChange function to begin with, but it wouldn't update unfortunately. The solution in that case is to allow max chars+1 and when they enter that extra key show the error, but that feels less responsive.

Reply:
Not sure I understand your point. Have you tried the proposed solution? I tested it and it works better than what was there, handling all cases (even the CTRL & other copy-paste methods).